app/assets/javascripts/discourse/app/routes/discovery-edit-category.js

```diff
@@ -4,12 +4,17 @@ import Category from "discourse/models/category";
 
 export default DiscourseRoute.extend({
   model(params) {
-    return Category.reloadBySlugPath(params.slug).then((result) => {
-      const record = this.store.createRecord("category", result.category);
-      record.setupGroupsAndPermissions();
-      this.site.updateCategory(record);
-      return record;
-    });
+    // don't reload model when switching tabs
```

Comment:
It would likely be simpler if you set it up like the user's route:

- category edit route
  - general
  - settings
  - etc

That way you don't need to do any logic when transitioning between child routes. The model is in the category-edit route and loads before you enter any of the children.

Reply:
Thanks for the suggestion, I have implemented it now. 

I found it tricky to set up the routing correctly inside the discovery route, so I pulled the "edit category" routes out of there.